Remember user's peer sort preference

jarolrod opened the pull request:
This sets the sort preference for the peers table in qsettings so that
it can be remembered. Also addresses an issue where the `id` toggle button is always checked when you go back to the table even though the table is not sorted by `id`.

[![Build Artifacts](https://img.shields.io/badge/Build%20Artifacts-green
)](https://github.com/bitcoin-core/gui-qml/actions/runs/6819663186)

johnny9 commented on src/qml/pages/node/Peers.qml:
```diff
@@ -17,6 +18,15 @@ Page {
 
+    Settings {
+        id: settings
+        property string lastSortBy: "nodeId"
```
this setting name might be too generic

jarolrod commented:
Updated from [979a984 ](https://github.com/bitcoin-core/gui-qml/commit/979a9842423980c94a1add4d4c0fade9f5339cfe)to [afcb1f7](https://github.com/bitcoin-core/gui-qml/commit/afcb1f7b9289b9b1f23491073ef84d0f7f353abc)

changes: made qsettings setting name for the last sort in the peer list less generic

johnny9 reviewed (approved):
ACK afcb1f7b9289b9b1f23491073ef84d0f7f353abc

hebasto commented:
Please rebase.

pablomartin4btc reviewed (approved):
tACK afcb1f7b9289b9b1f23491073ef84d0f7f353abc

jarolrod commented:
Updated from [afcb1f7 ](https://github.com/bitcoin-core/gui-qml/commit/afcb1f7b9289b9b1f23491073ef84d0f7f353abc)to [9cd558e](https://github.com/bitcoin-core/gui-qml/commit/9cd558e17a9a21c979dae9fa7274bf066d15effe)

Changes: rebased over main
